parquet/api.py

```diff
@@ -0,0 +1,280 @@
+from __future__ import unicode_literals
+
+import numpy as np
+import os
+import pandas as pd
+import re
+import struct
+import thriftpy
+
+from .core import ParquetException, read_thrift
+from .thrift_structures import parquet_thrift
+from .writer import write
+from . import core, schema, converted_types, encoding
+
+
+def def_open(f):
+    return open(f, 'rb')
+
+
+class ParquetFile(object):
+    """For now: metadata representation
+
+    Parameters
+    ----------
+    fname: path/URL string
+    verify: test file start/end bytes
+    open_with: function returning an open file
+    """
+
+    def __init__(self, fname, verify=False, open_with=def_open):
+        if isinstance(fname, str):
+            try:
+                # backend may not have something equivalent to `isdir()`
+                f = open_with(fname)
+            except (IOError, OSError):
+                fname = os.path.join(fname, '_metadata')
+                f = open_with(fname)
+        else:
+            f = fname
+            self.fname = str(fname)
+        self.open = open_with
+        self.fname = fname
+        self.f = f
+        self._parse_header(f, verify)
+        self._read_partitions()
+
+    def _parse_header(self, f, verify=True):
+        try:
+            f.seek(0)
+            if verify:
+                assert f.read(4) == b'PAR1'
+            f.seek(-8, 2)
+            head_size = struct.unpack('<i', f.read(4))[0]
+            if verify:
+                assert f.read() == b'PAR1'
+        except (AssertionError, struct.error):
+            raise ParquetException('File parse failed: %s' % self.fname)
+
+        f.seek(-(head_size+8), 2)
+        try:
+            fmd = read_thrift(f, parquet_thrift.FileMetaData)
+        except thriftpy.transport.TTransportException:
+            raise ParquetException('Metadata parse failed: %s' %
+                                         self.fname)
+        self.fmd = fmd
+        self.head_size = head_size
+        self.version = fmd.version
+        self.schema = fmd.schema
+        self.row_groups = fmd.row_groups or []
+        self.key_value_metadata = fmd.key_value_metadata
+        self.created_by = fmd.created_by
+        self.group_files = {}
+        for i, rg in enumerate(self.row_groups):
+            for chunk in rg.columns:
+                self.group_files.setdefault(i, set()).add(chunk.file_path)
+
+    @property
+    def columns(self):
+        return [f.name for f in self.schema if f.num_children is None]
+
+    def _read_partitions(self):
+        cats = {}
+        for rg in self.row_groups:
+            for col in rg.columns:
+                partitions = re.findall("([a-zA-Z_]+)=([^/]+)/",
+                                        col.file_path or "")
+                for key, val in partitions:
+                    cats.setdefault(key, set()).add(val)
+        self.cats = {key: list(v) for key, v in cats.items()}
+
+    def to_dask(self, columns=None, usecats=None, **kwargs):
```

Comment:
Would prefer `to_dask_dataframe`

Is `usecats` a standard term in Parquet?  If not then thoughts on `categories` or `use_categories`?